src/app/modules/verification/__tests__/verification.routes.spec.ts

```diff
@@ -0,0 +1,359 @@
+    }),
+  },
+}))
+
+describe('verification.routes', () => {
+  let mockTransaction: IVerificationSchema
+  let mockTransactionId: string
+  let mockEmptyFormId: string
+  let mockVerifiableFormId: string
+  let mockEmailFieldId: string
+  let mockMobileFieldId: string
+  let request: Session
+
+  beforeAll(async () => await dbHandler.connect())
+
+  beforeEach(async () => {
+    request = session(verificationApp)
+    jest.clearAllMocks()
+    await dbHandler.clearDatabase()
+
+    // Form without verifiable fields
+    const { form: emptyForm } = await dbHandler.insertEmailForm()
+    mockEmptyFormId = String(emptyForm._id)
+
+    // Form with verifiable fields
+    const emailField = generateDefaultField(BasicField.Email, {
+      isVerifiable: true,
+    })
+    const mobileField = generateDefaultField(BasicField.Mobile, {
+      isVerifiable: true,
+    })
+    mockEmailFieldId = String(emailField._id)
+    mockMobileFieldId = String(mobileField._id)
+    const { form: verifiableForm } = await dbHandler.insertEmailForm({
+      // Alternative mail domain so as not to clash with emptyForm
+      mailDomain: 'test2.gov.sg',
+      formOptions: {
+        form_fields: [emailField, mobileField],
+      },
+    })
+    mockVerifiableFormId = String(verifiableForm._id)
+
+    // Mock transaction with both email and mobile fields
+    mockTransaction = await VerificationModel.create({
+      formId: mockVerifiableFormId,
+      fields: [
+        generateFieldParams({
+          _id: mockEmailFieldId,
+          // Hash created 1 minute ago, so we can test requesting for new OTP
+          // without being rejected due to minimum waiting time
+          hashCreatedAt: new Date(Date.now() - 60 * 1000),
+          fieldType: BasicField.Email,
+          hashRetries: 0,
+          hashedOtp: MOCK_HASHED_OTP,
+          signedData: MOCK_SIGNED_DATA,
+        }),
+        generateFieldParams({
+          _id: mockMobileFieldId,
+          // Hash created 1 minute ago, so we can test requesting for new OTP
+          // without being rejected due to minimum waiting time
+          hashCreatedAt: new Date(Date.now() - 60 * 1000),
+          fieldType: BasicField.Mobile,
+          hashRetries: 0,
+          hashedOtp: MOCK_HASHED_OTP,
+          signedData: MOCK_SIGNED_DATA,
+        }),
+      ],
+    })
+    mockTransactionId = String(mockTransaction._id)
+  })
+
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('POST /', () => {
+    const ROUTE = '/transaction'
+
+    it('should return 400 when formId is not provided in body', async () => {
+      const response = await request.post(ROUTE)
+
+      expect(response.status).toBe(StatusCodes.BAD_REQUEST)
+      expect(response.body).toEqual(
+        buildCelebrateError({ body: { key: 'formId' } }),
+      )
+    })
+
+    it('should return 400 when formId is malformed', async () => {
+      const response = await request.post(ROUTE).send({ formId: 'malformed' })
```

Comment:
...

Reply:
oh this wasn't meant to be a joke but I guess it's still funny